### DISCUSSION TITLE
Unable to pass image to model using fastapi

### DISCUSSION THREAD

**Discussion by sparshgarg23** [fastapi/fastapi#6379] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, UploadFile, File
import uvicorn

import torch
import torchvision
from torchvision import transforms

from PIL import Image
from build_effnet import build_name_model
import torch.nn.functional as F

import io
from io import BytesIO
import numpy as np
import cv2

app = FastAPI()


class ClassificationServer:
    def __init__(self):
        self.class_names=['F_AF', 'F_AS', 'F_CA', 'F_LA', 'M_AF', 'M_AS', 'M_CA', 'M_LA']
        self.idx_to_class={i:j for i,j in enumerate(self.class_names)}
        self.class_to_idx={value:key for key,value in self.idx_to_class.items()}
        self.image_transforms={"test":transforms.Compose([
            transforms.oTensor(),
            transforms.Normalize([0.485,0.456,0.406],[0.229,0.224,0.225])
            ])}
        self.device=torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
        self.model=build_name_model()
        self.model.load_state_dict('CharacterClass_effnet_SGD.pt',map_location='cpu')
        self.model.eval()
        self.model.to(self.device)
    def read_image(self,image_payload):
        
        stream=BytesIO(image_payload)
        image=np.asarray(bytearray(stream.read()),dtype="uint8")
        image=cv2.imdecode(image,cv2.IMREAD_COLOR)
        
        if (isinstance(image,np.ndarray)):
            img=Image.fromarray(image)
        return img
    
    def classify(self,payload):
        img=self.read_image(payload)
        img=self.image_transforms["test"](img)
        
        with torch.no_grad():
            ps=self.model(img.unsqueeze(0))
            ps=F.softmax(ps,dim=1)
            topk,topclass=ps.topk(1,dim=1)
        x=topclass.view(-1).cpu().view()
        return self.idx_to_class[x[0]]
    
    @app.get("/hello")
    def get(self):
        return "Welcome to Image classfiication server."
    @app.post("/classify_image")
    async def classify_image(self,file:UploadFile=File(...)):
        image_bytes=await file.read()
        #return self.classify(image_bytes)
if __name__=="__main__":
    uvicorn.run(app,host="127.0.0.1",port=8000)
```

### Description

On executing the code,when I pass image to the server via curl command as shown below
```
curl -L -F "file=@model_0_0.jpg" http://127.0.0.1:8000/classify_image
```
I end up recieving the error
```
{"detail":[{"loc":["query","self"],"msg":"field required","type":"value_error.missing"}]}
```
Plus,I am unable to upload the image to the server.it seems that there is an issue with passing the file.Am I writing the code correctly.
I tried testing the upload funcitonality using the following code
```
from fastapi import FastAPI, File, UploadFile
import uvicorn

app = FastAPI()


@app.get("/files/")
async def create_file(file: bytes = File(...)):
    return {"file_size": len(file)}


@app.post("/uploadfile/")
async def create_upload_file(file: UploadFile):
    return {"file_size": file.filename}
if __name__=="__main__":
    uvicorn.run(app,host="127.0.0.1",port=8000)
```
But when I click on the server 127.0.0.1:8000 I end up getting method not allowed.Passing the file to server via curl returns its filename

Most of the docs end up using the same snippet,I also pointed the upload issue in previous topic.but this is the main task I am trying to solve,which is upload image to server,and then get model predictions.

Any help will be appreciated 

### Operating System

Windows

### FastAPI Version

0.75.1

### Python Version

3.9.7

### Additional Context

![fastapi_model_prediction](https://user-images.githubusercontent.com/34626942/165096325-d4a585b1-4d97-48cf-ba66-6d95843ac549.png)
![curl_command_fastapi](https://user-images.githubusercontent.com/34626942/165096350-2a2f22db-1074-4867-a5b7-b5de56807ec4.png)

**Comment by arunppsg** [comment, marked as answer]
In your code, the `get` and `classify_image` methods are inside the class which I think should not be. I made a minimal example which seemed to work:
```
 # code omitted
app = FastAPI()


class ClassificationServer:
    def __init__(self):
       # code omitted
        pass

    def read_image(self, image_payload):
        # code omitted
        stream = BytesIO(image_payload)
        image = np.asarray(bytearray(stream.read()),dtype="uint8")
        return image

    def classify(self, img):
        # code omitted
        return 1

@app.get("/hello")
def get():
    return "Welcome to Image classfiication server."

@app.post("/classify_image")
async def classify_image(file: UploadFile=File(...)):
    image_bytes = await file.read()
    classifier = ClassificationServer()
    img = classifier.read_image(image_bytes)
    return classifier.classify(img)

if __name__=="__main__":
    uvicorn.run(app,host="127.0.0.1",port=8000)
```

Visit `localhost:8000/docs` to access api endpoints.

**Comment by sparshgarg23** [comment]
thanks @arunppsg  it worked
